torch_geometric/datasets/ba_shapes.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional, Callable
+
+import torch
+import numpy as np
+import networkx as nx
+from torch_geometric.data import InMemoryDataset, Data
+
+
+def house():
+    G = nx.house_graph()
+    labels = [1, 1, 2, 2, 3]
+    attr = {i: labels[i] for i in range(5)}
+    nx.set_node_attributes(G, attr, 'label')
+    nx.set_edge_attributes(G, 1, 'label')
+    return G
+
+
+def ba(num_nodes=300, m=5, seed=None):
```

Comment:
There also exists [`barabasi_albert_graph`](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.barabasi_albert_graph) in PyG.